backend-peer/src/lib/services/cloudagent.ts

```diff
@@ -61,6 +71,33 @@ export class CloudagentManager {
     return agentInfo as AgentInfo
   }
 
+  receiveImplicitInvitation = async (did: string) => {
+    const requestBody = {
+      did,
+      handshakeProtocols: ['https://didcomm.org/connections/1.0'],
+      autoAcceptConnection: true,
+    }
+
+    const res = await fetch(
+      `${this.url_prefix}/oob/receive-implicit-invitation`,
+      {
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(requestBody),
+        method: 'POST',
+      }
+    )
+
+    const responseBody = await res.json()
+
+    if (res.status === 500) {
+      throw new BadRequest(responseBody.message)
```

Comment:
if you give a `did` that fails to resolve, the cloudagent 500s - which seems incorrect to me since a 500 normally represents unexpected behaviour and giving a bad `did` is quite likely. I thought it would be helpful for users of `nice-agent-portal` to get a response saying 'failed because of the did` rather than `generic 500`. That said we can't guarantee that all cloudagent 500s are because of that

Reply:
changed to `500` if the response from cloudagent is not OK in anyway

Reply:
in that case, your error is the correct one 👍

Reply:
changed to `500` if the cloudagent responds `500`*